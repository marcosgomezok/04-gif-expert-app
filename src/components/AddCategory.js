import React,{useState} from 'react'
import PropTypes from 'prop-types'
export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState(''); //useState () tiene un error de indefinido si no recibe nada, usar ''
    const handleInputChange = (e) => {
       // console.log(e.target.value)
        setInputValue(e.target.value)
        console.log('handleInputChange llamado')
    }

    const handleSubmit = (e) =>{
        e.preventDefault();
        console.log('handleSubmit',inputValue);

        if(inputValue.trim().length>2){
            setCategories(cats =>[ inputValue,...cats]);
            setInputValue('');
        }

        
    }
    
    return (
    
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
        <input 
            type="text"
            value={inputValue}
            onChange={handleInputChange}
            />
        </form>    
 
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}