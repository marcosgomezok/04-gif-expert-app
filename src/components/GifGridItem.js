import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({id,title,url}) => {

    //console.log(id,title,url);
    return (
        <div className='card animate__animated animate__fadeIn'>
          <img src={url} alt={title}/>
             <p>{title}</p>
        </div>
    )
}
//  {img.title} 

/* 1. Enzyme
   2. Enzyme to json
   3. debe de mostrar el componente correcto
    shallow
    wrapper
    wrapper .toMatchSnapshot()
*/

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  
}