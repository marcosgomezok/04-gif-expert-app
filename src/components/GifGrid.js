//import React,{useState, useEffect} from 'react'
//import { getGifs } from '../helpers/getGifs'; 
//el import de arriba se usa en el custom hook
import { GifGridItem } from './GifGridItem';
import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {


//    const [images, setImages] = useState([]);
//    const state = useFetchGifs();
    const {data:images,loading} = useFetchGifs(category);


    return (
        <>
        <h3 className="animate__animated animate__fadeIn">{category}</h3>
        {loading && <p>Loading</p>}
        <div className="card-grid animate__animated animate__flash">

            {
                images.map(img => (
                <GifGridItem key={img.id}
                {...img}
                />
                ))
            }
        </div> 
        </>
    )
}
GifGrid.propTypes ={
    category: PropTypes.string.isRequired
}

    //const [count, setCount] = useState(0)

    // getGifs();
/*

    useEffect(()=>{
        getGifs(category)
            .then(setImages);
    },[category]) */

//.then(imgs => setImages(imgs))


/*     const getGifs = async() =>{

   

        const url = `https://api.giphy.com/v1/gifs/search?q=${encodeURI(category)}&limit=10&api_key=dQaadwLZwFULzd2dI2vMmTfZdGI3SMuj`
        
        const resp = await fetch(url);
        const {data} = await resp.json();

        const gifs = data.map(img =>{
            return{
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url,
            }
        })
        console.log(gifs) 
        setImages(gifs);
    } */

    /* <h3>{count}</h3>
<button onClick={ () => setCount ( count + 1 ) }></button> */

//images.map(img => (
//sin desestructuracion linea 53
/* 
<ol>
{
    images.map(({id,title}) => (
                <li key={id}>{title}</li>
    ))
}
</ol>
 */

 /* <div className="card-grid">

            {
                images.map(img => (
                <GifGridItem key={img.id}
                {...img}
                />
                ))
            }
        </div> */