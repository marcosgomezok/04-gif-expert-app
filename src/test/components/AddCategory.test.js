import { shallow } from 'enzyme'
import React from 'react'
import { Simulate } from 'react-dom/test-utils';
import { AddCategory } from '../../components/AddCategory';
import '@testing-library/jest-dom'

describe('Pruebas en <AddCategory />', () => {
    
    const setCategories= jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);


    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    });
    test('debe de mostrarse correctamete', () => {
        
       
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de cambiar la caja de texto', () => {
        
        const input = wrapper.find('input');
        
        const value = 'Hola Mundo'
        input.simulate('change',{target:{value}});

        expect(wrapper.find('p').text().trim()).toBe(value)

    })
    
    test('No debe de postear la informacion con submit', () => {
        
        wrapper.find('form').simulate('submit',{preventDefault:()=>{}});

        expect(setCategories).not.toHaveBeenCalled();




    })

    test('debe de llamar el setCategories y limpiar la caja de texto', () => {
        
        
        // 1.simular el inputChange
        const value = 'Hola Mundo'
        wrapper.find('input').simulate('change',{target:{value}});

        // 2.simular el submit

        wrapper.find('form').simulate('submit',{preventDefault(){}});

        // 3.setCategories se debe de haber llamado
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

        //4. el valor del input debe de estar ''
        expect(wrapper.find('input').prop('value')).toBe('');

    })
    
    
    
})
