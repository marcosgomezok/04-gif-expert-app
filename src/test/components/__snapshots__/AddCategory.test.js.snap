// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe de mostrarse correctamete 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
